quic/s2n-quic/src/tests/dc.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::*;
+use s2n_quic_core::{
+    dc::testing::MockDcEndpoint,
+    event::{api::DcState, Timestamp},
+    stateless_reset::token::testing::{TEST_TOKEN_1, TEST_TOKEN_2},
+};
+
+// Client                                                                    Server
+//
+// Initial[0]: CRYPTO[CH (DC_SUPPORTED_VERSIONS[3,2,1]] ->
+//
+//                                      # dc_state_changed: state=VersionNegotiated
+//                                                    Initial[0]: CRYPTO[SH] ACK[0]
+//                Handshake[0]: CRYPTO[EE (DC_SUPPORTED_VERSIONS[3], CERT, CV, FIN]
+//
+// # dc_state_changed: state=VersionNegotiated
+// # handshake_status_updated: status=Complete
+// # dc_state_changed: state=PathSecretsReady
+// Initial[1]: ACK[0]
+// Handshake[0]: CRYPTO[CERT, CV, FIN], ACK[0] ->
+// 1-RTT[0]: DC_STATELESS_RESET_TOKENS[..]
+//
+//                                       # dc_state_changed: state=PathSecretsReady
+//                                      # handshake_status_updated: status=Complete
+//                                     # handshake_status_updated: status=Confirmed
+//               <- 1-RTT[1]: HANDSHAKE_DONE, ACK[0], DC_STATELESS_RESET_TOKENS[..]
+//
+// # handshake_status_updated: status=HandshakeDoneAcked
+// # handshake_status_updated: status=Confirmed
+// # key_space_discarded: space=Handshake
+// # dc_state_changed: state=Complete
+//
+// 1-RTT[1]: ACK[1] ->
+//                            # handshake_status_updated: status=HandshakeDoneAcked
+//                                               # dc_state_changed: state=Complete
+#[test]
+fn dc_mtls_handshake_self_test() {
```

Comment:
technically you can do a dc connection without an MTLS, right? we should probably have tests for that as well